Add env var overrides for controller enable flags (#3181)
DataDog/datadog-operator: docs/installation.md

| Secure metrics             | `--metrics-secure`                   | `DD_METRICS_SECURE`                   | `false` |
| Profiling                  | `--profiling-enabled`                | `DD_PROFILING_ENABLED`                | `false` |
| Pprof                      | `--pprof`                            | `DD_PPROF_ENABLED`                    | `false` |
| Leader election            | `--enable-leader-election`           | `DD_LEADER_ELECTION_ENABLED`          | `true`  |
| Leader election lease      | `--leader-election-lease-duration`   | `DD_LEADER_ELECTION_LEASE_DURATION`   | `60s`   |
| Cilium network policies    | `--supportCilium`                    | `DD_SUPPORT_CILIUM`                   | `false` |
| Maximum goroutines         | `--maximumGoroutines`                | `DD_MAXIMUM_GOROUTINES`               | `400`   |
| Controller revisions       | `--createControllerRevisions`        | `DD_CREATE_CONTROLLER_REVISIONS`      | `false` |

ExtendedDaemonset options (`--supportExtendedDaemonset` and `--eds*`) and
log options (`--loglevel`, `--logEncoder`) and secret backend options
(`--secretBackend*`, `--secretRefreshInterval`) and `--version` are only

> configurable via CLI flag.

Reviewer: ```suggestion
ExtendedDaemonset options (`--supportExtendedDaemonset` and `--eds*`),
log options (`--loglevel`, `--logEncoder`), secret backend options
(`--secretBackend*`, `--secretRefreshInterval`), and `--version` are only
configurable using CLI flags.
```

Boolean values follow Go's [`strconv.ParseBool`](https://pkg.go.dev/strconv#ParseBool):
`true`, `True`, `TRUE`, `1` or `false`, `False`, `FALSE`, `0`. The strings

> `yes` and `no` are **not** accepted and will be logged as an error, leaving the

Reviewer: ```suggestion
`yes` and `no` are **not** accepted and are logged as errors, leaving the
```

default in effect.

Duration values follow Go's [`time.ParseDuration`](https://pkg.go.dev/time#ParseDuration),

> for example `30s`, `5m`, or `1h`. Integer values use base 10.

Reviewer: ```suggestion
Duration values follow Go's [`time.ParseDuration`](https://pkg.go.dev/time#ParseDuration);
for example, `30s`, `5m`, or `1h`. Integer values use base 10.
```
